draft-private-access-tokens.md

```diff
@@ -173,14 +185,83 @@ Mediators advertise an Oblivious HTTP URI template {{!RFC6570}} for proxying
 protocol messages to Issuers. For example, one template
 for the Mediator might be https://mediator.net/relay-access-token-request.
 
-# Issuance
+# Protocol
+
+Private Access Tokens are single-use tokens cryptographically bound to
+policies. Redeemers request tokens from Clients, who then engage with
+Mediators and Issuers to private compute policy-compliant tokens and
+reveal them to the Redeemer. Example policies and use cases that system
+addresses are described in {{examples}}.
+
+The rest of this section describes this interactive protocol in terms of
+the redemption request ({{redemption-request}}) and corresponding token
+issuance flow ({{issuance}}).
+
+## Redemption Request
+
+The Client is assumed to have the policy verification key before redeeming
+a Private Access Token. See {{access-token-keys}} for details on how this
+can be obtained.
+
+Token redemption is an interactive protocol. Redeemers challenge Clients to
+present a unique, single-use token. Redeemers present this challenge to Clients
+with the following context:
+
+~~~
+struct {
+   uint8 policy_window;
+   opaque redeemer_name<1..2^16-1>;
+   opaque issuer_name<1..2^16-1>;
+   opaque redemption_nonce[32];
+   uint64 timestamp;
+} RedemptionContext;
+~~~
+
+policy_window:
+: The policy window value for the redemption token (ISSUER_POLICY_WINDOW).
+
+redeemer_name:
+: Name of the Redeemer (ORIGIN_NAME).
+
+issuer_name:
+: Name of the Issuer (ISSUER_NAME).
+
+redemption_nonce:
+: A fresh 32-byte nonce generated for each redemption request.
+
+timestamp:
+: The current UNIX epoch in seconds.
+
+This context is sent to Clients in an "WWW-Authenticate" header as follows:
+
+~~~
+WWW-Authenticate: PrivacyToken realm="<context>"
```

Comment:
I'm not sure the context should be stuck in the "realm" variable. Can we have a new variable here? And we should likely clarify it needs to be a quoted base64 string.

Reply:
Maybe just call it "challenge" or "context". Though if the redemption origin is in the context, we don't need a separate realm?

Reply:
I'd like to copy this: https://www.rfc-editor.org/rfc/rfc7486.html#section-3

Reply:
Yeah, "challenge" is good!